### PR TITLE
Feature optimizations and refactorings for anime route

### PR DESCRIPTION
**Pull request by delikesance**
This merge includes a series of significant changes aimed at optimizing the `/animes` route, introducing caching mechanisms, enhancing query handling, and refining the API's structure for better performance and maintainability. Major features and refactorings are highlighted below:

- **feat(animes-route):** Optimized anime list fetching with caching and improved query handling, introducing a cache-first strategy to enhance performance significantly. Adjusted query parameter handling for filtering animes by title, genres, status, and release date to ensure more efficient database queries. Also included streamlined handling for fetching detailed anime information when necessary and adjusted the main application logger settings for development purposes. Note that these changes might affect existing queries due to the adjustments in query parameter handling, marking a breaking change.

- **refactor:** Various refactorings across the codebase to improve code quality and organization. This includes marking plugin and route default exports as async, improving query construction with better type safety, and removing redundant comments in favor of relying on TypeScript's typing system and external API documentation.

- **feat:** Introduced and enhanced several services (`AnimeService`, `UserService`) for centralized data handling and operations related to anime data and user registration/authentication, respectively.

- **BREAKING CHANGE:** Transitioned from using Prettier and ESLint to Biome for code formatting and linting to align with new development standards.

These changes are expected to improve the scalability, performance, and maintainability of the gazes-api, setting a solid foundation for future enhancements.

PR by Angelo Clauin <contact@l31.dev>

### CODE REVIEW

**Review by ncs-pl — COMMENTED** [review]
lgtm, appart from the already existing issues from automatic static analysis.
I would also encourage you avoiding "magic values", and instead declare constants that are being inlined at compile time, for illustration:

```ts
response.status(200)...;
```

to be changed to

```ts
const HTTP_OK = 200;

response.status(HTTP_OK)...;
```

File: apps/gazes-api/src/app/routes/animes.ts [hunk 1]
```diff
@@ -1,56 +1,57 @@
...
 
-		if (title) queryOptions.where["others"] = { search: title.split(" ").join(" & ") };
-		if (genres) queryOptions.where["genres"] = { hasEvery: genres.split(",") };
-		if (status) queryOptions.where["status"] = { equals: status.toString() };
-		if (releaseDate)
-			queryOptions.where["start_date_year"] = {
-				equals: releaseDate.toString(),
-			};
+		let animeList = await prisma.anime.findMany(queryOptions);
+		await redis.setEx(cacheKey, 3600, JSON.stringify(animeList));
```
Comment (lilmaax) [inline apps/gazes-api/src/app/routes/animes.ts:41]: What does "3600" value mean? Maybe you can set an explicit variable instead

**Review by maxchrr — COMMENTED** [review]
lgtm, Code Quality analysis has already done all the job

File: apps/gazes-api/src/app/plugins/jwt.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
...
+ * This plugins adds some utilities to handle http errors
+ *
+ * @see https://github.com/fastify/fastify-sensible
+ */
+export default fp(async (fastify: FastifyInstance) => {
+	fastify.register(jwt, {
+        secret: getEnv("JWT_SECRET")
+    });
+    
+});
```
Comment (lilmaax) [inline apps/gazes-api/src/app/plugins/jwt.ts:16]: Indentation seems weird

File: apps/gazes-api/src/app/routes/animes.ts [hunk 3]
```diff
@@ -1,93 +1,49 @@
...
-		try {
-			const cachedLatestList = await redis.get("latests");
-
-			if (cachedLatestList) {
-				rep.status(200).send(JSON.parse(cachedLatestList));
-				return;
+import { AnimeDetailParams, AnimeDetailParamsSchema, AnimeListQuerystring, AnimeListQuerystringSchema } from "@api/contracts/animesContract";
+import { StatusCodes } from "http-status-codes";
+import { Anime, Prisma } from "@prisma/client";
+import { CacheManager } from "../utils/cacheManager";
```
Comment (lilmaax) [inline apps/gazes-api/src/app/routes/animes.ts:7]: Try not to use relative import, if you move the file or implementation, you can break it

File: apps/gazes-api/src/app/routes/animes.ts [hunk 4]
```diff
@@ -98,48 +54,6 @@ export default async function (fastify: FastifyInstance, { redis, prisma }: AppO
 
 		rep.status(200).send(anime);
```
Comment (lilmaax) [inline apps/gazes-api/src/app/routes/animes.ts:55]: Missing `StatusCodes` variable

File: apps/gazes-api/src/app/routes/auth.ts [hunk 5]
```diff
@@ -24,7 +24,7 @@ export default function (fastify: FastifyInstance, { prisma }: AppOptions) {
 	fastify.post<{ Body: RegisterBody }>("/register", { schema: { body: RegisterBodySchema } }, async (req, rep) => {
 		try {
 			const { email, username, password } = req.body;
-			const { user, token } = await userService.registerUser(email, username, password);
+			const { user, token } = await userService.registerUser(email, username, password, fastify);
 
 			rep.header("Set-Cookie", `token=${token}; HttpOnly; Path=/; Secure; SameSite=Strict`);
 			rep.status(201).send(user);
```
Comment (lilmaax) [inline apps/gazes-api/src/app/routes/auth.ts:30]: Missing `StatusCodes` variable

File: apps/gazes-api/src/app/routes/auth.ts [hunk 6]
```diff
@@ -50,7 +50,7 @@ export default function (fastify: FastifyInstance, { prisma }: AppOptions) {
 	fastify.post<{ Body: LoginBody }>("/login", { schema: { body: LoginBodySchema } }, async (req, rep) => {
 		try {
 			const { email, password } = req.body;
-			const { user, token } = await userService.authenticateUser(email, password);
+			const { user, token } = await userService.authenticateUser(email, password, fastify);
 
 			rep.header("Set-Cookie", `token=${token}; HttpOnly; Path=/; Secure; SameSite=Strict`);
 			rep.status(200).send(user);
```
Comment (lilmaax) [inline apps/gazes-api/src/app/routes/auth.ts:56]: Missing `StatusCodes` variable

File: apps/gazes-api/src/app/services/animeService.ts [hunk 7]
```diff
@@ -1,123 +1,109 @@
...
+	private buildQueryOptions({
+		page,
+		title,
+		genres,
+		status,
+		releaseDate,
+	}: { page: number; title: string; genres: string; status: number; releaseDate: number }): Prisma.AnimeFindManyArgs {
+		return {
+			skip: 25 * (page - 1),
+			take: 25,
```
Comment (lilmaax) [inline apps/gazes-api/src/app/services/animeService.ts:87]: Maybe you can make this more readable, or explain it with comment

File: apps/gazes-api/src/main.ts [hunk 8]
```diff
@@ -1,46 +1,94 @@
...
 async function main() {
-	await redis.connect();
+	const prismaClient = new PrismaClient();
+	const redisClient = await connectToRedis();
 
-	if (process.env.NODE_ENV === "development") {
-		redis.flushDb();
-	}
+	updateAnimeDatabase(prismaClient);
+	setInterval(() => updateAnimeDatabase(prismaClient), 3600000);
```
Comment (lilmaax) [inline apps/gazes-api/src/main.ts:21]: Use constant instead of set value, and explain what it means

**Review by maxchrr — COMMENTED** [review]
More generally, there seem to be indentation/formatting issues in a few lines, which is very weird.

**Review by maxchrr — COMMENTED** [review]
LGTM

Don't forget to add comments, everything is very rough, to help you understand the code.

There also seem to be some issues with the rule for spaces between braces, so try to be consistent.

**Pull request merged**